### DISCUSSION TITLE
Regex for input field that allows comma, dot and hyphen, but needs at least one letter

### DISCUSSION THREAD

**Discussion by AstridOttmann** [neuefische/questions#466] [Web]
I tried it like this:  
pattern="^.[^0-9]{1,30}"
but so you need two enter at least 2 characters (it should work with one) and it also allows only whitespace (it should need at least one letter)

**Comment by doemser** [comment]
No one likes regex so I asked AI for that regex task which I am not abled to validate myself. Try if this helps your task:

```js
^(?=.*[a-zA-Z])[a-zA-Z0-9,.-]*$

```

This regular expression will match any string that contains at least one letter and may also contain numbers, commas, dots, and hyphens. The ^ and $ anchor the pattern to the start and end of the string, respectively, and the (?=.*[a-zA-Z]) is a positive lookahead that ensures at least one letter is present.

**Reply by AstridOttmann** [reply]
Sorry, it's not working! It does not accept anything now.

**Reply by doemser** [reply]
What are example words you want to match? And what are example words you don't want to match? 

Als try this:
[a-zA-Z][a-zA-Z0-9,.-]*

Now every input has to start with a letter a-z or A-Z and can be followed by every number and ,.-

**Reply by doemser** [reply]
However, the use-case is not yet clear to me. What do you want to validate with it?

**Reply by AstridOttmann** [reply]
I want to be able to insert one or more letters, no numbers, but comma, dot, whitespace and hyphen (language word input).
It should not accept only whitespace and more than 30 characters

**Reply by AstridOttmann** [reply]
[a-zA-Z][a-zA-Z0-9,.-]*
this one is not accepting whitespace at all (you should be able to enter a sentence)

**Reply by doemser** [reply]
[a-zA-Z][a-zA-Z\s,.-]*

This should do what you are asking for now. BUT please test it with A LOT of examples from differen languages etc. to verify that this is really what you want to achieve

**Reply by AstridOttmann** [reply]
Almost there! 
![Screenshot 2023-01-25 at 1 56 56 PM](https://user-images.githubusercontent.com/118136613/214569133-d4c6d377-7634-4b7f-a015-18b4d31186b5.png)
I added äöü etc., that works, but one problem still there (screenshot). My solution did not work: 
[a-zA-ZäöüÄÖÜ][a-zA-ZäöüÄÖÜ\s,.-]{1,30}*

**Reply by doemser** [reply]
It is still not clear to me what you expect and what happens instead. maybe you just want to exclude some few signs, instead of including a lot of signs

**Comment by AstridOttmann** [comment]
I spent the whole day with this, because it was critized. Can I go on without a perfect solution? 

What I want is: 
- insert at least one letter (only spaces should throw an error, but only one letter has to be possible)
- no numbers
- spaces, comma, dot and hyphen
- not more than 30 characters (to avoid overflow)

**Comment by AstridOttmann** [comment]
pattern="[a-zA-ZäöüÄÖÜ][a-zA-ZäöüÄÖÜ\s,.-]*"
          maxLength="30"

this seems to work!

**Comment by doemser** [comment]
I like how you went from "can I skip this" to "now I have something that seems to work". 

There are a lot of things to consider when it comes to form validation, I think you've gone into it in enough detail and now know where to locate the issues.

**Reply by AstridOttmann** [reply]
My final solution: 
 pattern="^[^\s0-9].*$"

- excludes whitespace and numbers  [^\s0-9]
- at the beginning of the string ^[ ]
- allows any character (.)
- zero or more times (*)
- at the end of the string ($)
